### DISCUSSION TITLE
TCP Proxy

### DISCUSSION THREAD

**Discussion by Zat-Code** [Kong/kong#8371] [Help]
Hello ! 

I would like to be able to have a single presentation on the internet and that several services can connect to it via a tcp url and be redirected to the right server on my local machine: 

service A => kong => server A

I also have a rabbitmq service that needs to connect to a rabbitmq server.

service rabbitmq => kong => server rabbitmq

I have tried several things but nothing works. 

I set the environment variable KONG_STREAM_LISTEN: 127.0.0.1:8999 in order to receive my tcp stream on this port. 

For the rabbitmq service for example, I want to redirect it to localhost:5672, 

I have created the service :

![Screenshot from 2022-02-03 18-09-19](https://user-images.githubusercontent.com/33628286/152393511-cd56dd68-34fc-4131-9db3-512956b8fc7f.png)

and the routes :

![Screenshot from 2022-02-03 18-10-43](https://user-images.githubusercontent.com/33628286/152393586-6cab6164-d6e0-4939-9538-f52c7cbc00d2.png)

is this good? 

moreover i connect my service to :

url_server:8999/rabbitmq

Is this ok?

**Comment by hbagdi** [comment]
If you want to expose two services using TCP, you need two distinct ports for those two services.

**Reply by Zat-Code** [reply]
Ah ok, I wanted to use cong as a reverse proxy to expose only one port on my machine. 

So it's not possible to redirect an udp stream by url via cong?
